game-core/src/main/java/games/strategy/engine/framework/startup/mc/ServerModel.java

```diff
@@ -110,23 +107,12 @@ static RemoteName getObserverWaitingToStartName(final INode node) {
   private CountDownLatch removeConnectionsLatch = null;
   private final Observer gameSelectorObserver = (observable, value) -> gameDataChanged();
 
-  ServerModel(final GameSelectorModel gameSelectorModel, final SetupPanelModel typePanelModel) {
-    this.gameSelectorModel = gameSelectorModel;
-    this.typePanelModel = typePanelModel;
-    this.gameSelectorModel.addObserver(gameSelectorObserver);
-    headlessServerSetupPanelModel = null;
-    headless = false;
-  }
-
-
-  public ServerModel(
-      final GameSelectorModel gameSelectorModel,
-      final HeadlessServerSetupPanelModel headlessServerSetupPanelModel) {
-    this.gameSelectorModel = gameSelectorModel;
-    typePanelModel = null;
+  public ServerModel(final GameSelectorModel gameSelectorModel, final ServerSetupModel serverSetupModel,
+      final JFrame ui) {
```

Comment:
Nit: Might want to annotate both the `ui` parameter and field as `@Nullable` so it's clear this is usable from both headed and headless contexts.